src/sagas/wallet.js

```diff
@@ -531,10 +545,37 @@ export function* onWalletConnStateUpdate({ payload }) {
   yield put(isOnlineUpdate({ isOnline }));
 }
 
+export function* walletReloading() {
+  const wallet = yield select((state) => state.wallet);
+  const routerHistory = yield select((state) => state.routerHistory);
+
+  // Since we close the channel after a walletReady event is received,
+  // we must fork this saga again so we setup listeners again.
+  yield fork(listenForWalletReady, wallet);
+
+  // Wait until the wallet is ready
+  yield take(types.WALLET_STATE_READY);
+
+  try {
+    // Store all tokens on redux as we might have lost tokens during the disconnected
+    // period.
+    const { allTokens } = yield call(loadTokens);
+
+    // Load success, we can send the user back to the wallet screen
+    yield put(loadWalletSuccess(allTokens));
+    routerHistory.replace('/wallet/');
+    yield put(loadingAddresses(false));
+  } catch (e) {
+    yield put(startWalletFailed());
```

Comment:
In the `startWallet` method you have

```
yield cancel(threads);
```

Should we cancel any thread here as well?

We also have some different code after success

```
  yield take(types.START_WALLET_REQUESTED);
  yield cancel(threads);
```

Why are they different? Can we reuse the same code?

Reply:
> Should we cancel any thread here as well?

The wallet is not being initialized again, we are only re-downloading the token balances, so the `take` effect on the `startWallet` saga is still waiting for a `START_WALLET_REQUESTED` action to cancel the threads that still need to be running in background, so we don't have to do anything here

But I just noticed that we might have lost transactions during the downtime, so we should invalidate the token history for all tokens (except hathor that is already downloaded on the `loadTokens` saga)

Done in e03c9bfae103c50bf899995c28bf7b81a0e7be50

Reply:
By the way, this might happen on the wallet-mobile as well, we should also apply this commit to it

Reply:
Ok, did you create a KTLO for that?

Reply:
https://github.com/HathorNetwork/hathor-wallet-mobile/issues/200